src/webassets/bundle.py

```diff
@@ -744,7 +744,13 @@ def _urls(self, ctx, extra_filters, *args, **kwargs):
             if ctx.auto_build:
                 self._build(ctx, extra_filters=extra_filters, force=False,
                             *args, **kwargs)
-            return [self._make_output_url(ctx)]
+            if with_bodies:
+                # The hunks returned by the _build() method aren't useful as
+                # they sometimes have %(version)s in them. Can this be changed?
+                hunk = FileHunk(self.resolve_output(ctx))
```

Comment:
That amazing work, thanks a bunch for putting in the time. Unfortunately, I think we have a problem here that is possibly complex, which is that before this patch, you could set ``auto_build`` to False, and the correct urls could be generated without the filesystem being touched; indeed, the actual files needn't be available.

Of course, if ``ASSETS_CONTENT`` is used, that is no longer possible. We can also ignore the question whether webassets should cache the content in memory for now; but the problem is that right now, by always using ``urls_with_content`` in the template tag, unless I am mistaken, you are essentially forcing a file read in all cases.

Ideally, the tag could detect the presence of ``ASSETS_CONTENT`` itself. I'm not sure if that is possible.